File: python/turing.py
#!/usr/bin/python3

from turingMachineIO import turingMachineFromFormatString
from pathlib import Path

import sys

def main(argv:list[str]) -> int:
    if len(argv) < 3:
        print(f"Usage: python3 ./turing.py <format> <filepath> <initialTape>")
        return 1
    
    machine = turingMachineFromFormatString(argv[0], Path(argv[1]).read_text())
    machine.clear([symbol for symbol in argv[2]])

    print(f"Initial tape: {machine.tape()}")
    machine.run()

    print(f"Final tape: {machine.tape()}")
    print(f"Final state: {machine.currentState()}")
    return 0 if machine.halted() else 1

if __name__ == "__main__":
    returnCode = main(sys.argv[1::])
    exit(returnCode)

File: go/turing.go
package main

import (
	"fmt"
	"log"
	"os"
	"turing/interpreter"
)

func main() {
	if len(os.Args) < 4 {
		log.Fatalf("Usage: %s <format> <filename> <tape>", os.Args[0])
		log.Fatal("Available formats:")
		log.Fatal("   - toml: TOML file format")

		os.Exit(1)
	}

	initialTape := make([]interpreter.TuringSymbol, len(os.Args[3]))
	for index, symbol := range os.Args[3] {
		initialTape[index] = interpreter.TuringSymbol(symbol)
	}

	turingMachine := interpreter.LoadTuringMachinFromFile(os.Args[1], os.Args[2])
	//turingMachine.Print()
	turingMachine.Clear(initialTape)

	fmt.Printf("Initial tape: %s\n", turingMachine.GetTape())
	turingMachine.Run()
	fmt.Printf("Final tape: %s\n", turingMachine.GetTape())
	fmt.Printf("Final state: %s\n", turingMachine.GetCurrentState())

	os.Exit(0)
}
